elasticsearch/src/main/java/org/hibernate/search/elasticsearch/analyzer/ElasticsearchCharFilterFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.elasticsearch.analyzer;
+
+import org.apache.lucene.analysis.util.CharFilterFactory;
+import org.hibernate.search.annotations.AnalyzerDef;
+
+/**
+ * A token filter factory to use with {@link AnalyzerDef} annotations
+ * to define an Elasticsearch token filter.
+ * <p>
+ * <strong>Caution:</strong> parameter values are interpreted as JSON, though using lenient parsing
+ * (quotes around strings may be left out in some cases, as when a string only contains letters).
+ * <p>
+ * Example:
+ * <pre><code>
+ * {@literal @}CharFilterDef(
+ * 		name = "custom-pattern-replace",
+ * 		factory = ElasticsearchCharFilterFactory.class,
+ * 		params = {
+ * 				{@literal @}Parameter(name = "type", value = "pattern_replace"),
+ * 				{@literal @}Parameter(name = "pattern", value = "'[^0-9]'"),
+ * 				{@literal @}Parameter(name = "replacement", value = "'0'"),
+ * 				{@literal @}Parameter(name = "tags", value = "'CASE_INSENSITIVE|COMMENTS'")
```

Comment:
I agree it could do, but I'm not sure it would be better.

1. We'd have some magic parameter name (not really intuitive)
2. We'd have more error cases to handle (what happens if someone gives a parameter name other than the magic one?)
3. It would not be much more readable, thanks to Java lacking multi-line string constants (a real pain, that one...)
4. On top of that, I'm not even sure there are analyzers with "object" parameters in Elasticsearch. I sure didn't come across any. The parameters are essentially flat, with arrays sometimes. So we wouldn't gain much...

To be honest, I'd say "yes" if we were introducing Elasticsearch-specific annotations, but in that case it doesn't seem to fit the annotation very well...